src/DurableTask.Netherite/PartitionState/PrefetchState.cs

```diff
@@ -74,13 +74,19 @@ public override void Process(WaitRequestReceived waitRequestEvent, EffectTracker
         void ProcessClientRequestEventWithPrefetch(ClientRequestEventWithPrefetch clientRequestEvent, EffectTracker effects)
         {
             if (clientRequestEvent.Phase == ClientRequestEventWithPrefetch.ProcessingPhase.Read)
-            {           
-                this.Partition.Assert(!this.PendingPrefetches.ContainsKey(clientRequestEvent.EventIdString), "PendingPrefetches.ContainsKey(clientRequestEvent.EventIdString)");
-
-                // Issue a read request that fetches the instance state.
-                // We have to buffer this request in the pending list so we can recover it.
+            {
+                if (!this.PendingPrefetches.ContainsKey(clientRequestEvent.EventIdString))
+                        {
+                    // Issue a read request that fetches the instance state.
+                    // We buffer this request in the pending list so we can recover it, and can filter duplicates
+                    // (as long as the duplicates appear soon after the original)
```

Comment:
nit - I want to call out explicitly that EventHubs may duplicate client-to-partition events, which is implied by the original comment, but not made totally explicit in my opinion. Just trying to put it in words that would make sense to me :-) .
```suggestion
            {
                // It's possible for EventHubs to duplicate client-to-partition events. Therefore, we perform a best-effort
                // de-duplication of EH messages. For more details, see: https://github.com/microsoft/durabletask-netherite/pull/379
                if (!this.PendingPrefetches.ContainsKey(clientRequestEvent.EventIdString))
                        {
                    // Issue a read request that fetches the instance state.
                    // We buffer this request in the pending list so we can recover it, and can filter duplicates
                    // (as long as the duplicates appear soon after the original)
```